controllers/controller.js

```diff
@@ -1,30 +1,55 @@
-const TaskModel = require('../models/task');
+const TaskModel = require('../models/task')
 
 // find all tasks in the database
 module.exports.findAllTasks = async (req, res) => {
-  const tasks = await TaskModel.find();
-  res.json(tasks);
+  const tasks = await TaskModel.find()
+  res.json(tasks)
 }
 
 // add new task to database
 module.exports.addTask = async (req, res) => {
-  const tasks = await TaskModel.create(req.body)
-  res.json(tasks);
+  const title = req.body.title
+  const isCompleted = req.body.isCompleted
+  const createdAt = new Date().toLocaleString()
+  const updatedAt = ''
+  const tasksTobeAdded = new TaskModel({
+    title,
+    isCompleted,
+    createdAt,
+    updatedAt,
+  })
+
+  const tasks = await tasksTobeAdded.save()
+  return res.json(tasks)
 }
 
 // update tasks
 module.exports.updateTask = async (req, res) => {
-  await TaskModel.findByIdAndUpdate(req.params.id, req.body, (err, tasks) => {
-    if (err) return res.status(500).json(err);
-    return res.json(tasks);
-  });
+  const dataToUpdate = req.body
+  dataToUpdate.updatedAt = new Date().toLocaleString()
+  await TaskModel.findByIdAndUpdate(
+    req.params.id,
+    dataToUpdate,
+    (err, tasks) => {
+      if (err) return res.status(500).json(err)
+      return res.json(tasks)
+  })
 }
 
 // delete tasks
 module.exports.deleteTask = async (req, res) => {
   await TaskModel.findByIdAndRemove(req.params.id, (err, tasks) => {
-    if (err) return res.status(500).json(err);
+    if (err) return res.status(500).json(err)
     const message = 'Deleted task'
-    return res.json(message);
+    return res.json(message)
+  })
+}
+
+// clear completed tasks
+module.exports.deleteCompleted = async (req, res) => {
+  await TaskModel.deleteMany( {isCompleted: true}, (err, tasks) => {
+    if (err) return res.status(500).json(err)
+    const message = 'Clear completed tasks'
+    return res.json(message)
```

Comment:
As I told you, we must define the API as simple as possible. I don't think it's a good idea when you split the delete endpoint. If you want to delete only the completed task, please handle it on the front-end side.

Reply:
I fixed this mistake